test/test_swa.py

```diff
@@ -0,0 +1,692 @@
+import functools
```

Comment:
This file should stay as `test_optim.py`

Reply:
Ok, but note that some of the common code in the `test_optim.py` was modified specifically for the SWA syntax; see e.g. https://github.com/izmailovpavel/contrib/blob/master/test/test_swa.py#L60-L61. Is it ok?

Reply:
Uh ok, it's not perfect, but we can leave it then 😕